core/cap-0062.md

```diff
@@ -0,0 +1,281 @@
+```
+CAP: 0062
+Title: Soroban Live State Prioritization
+Working Group:
+    Owner: Garand Tyson <@SirTyson>
+    Authors: Garand Tyson <@SirTyson>
+    Consulted: Dmytro Kozhevin <@dmkozh>, Nicolas Barry <@MonsieurNicolas>
+Status: Draft
+Created: 2024-12-02
+Discussion: https://github.com/stellar/stellar-protocol/discussions/1575
+Protocol version: 23
+```
+
+## Simple Summary
+
+This proposal allows the network to evict `PERSISTENT` entries from the live state BucketList to a separate BucketList
+containing archival state. Note that evicted entries are not deleted from validators.
+
+## Working Group
+
+As specified in the Preamble.
+
+## Motivation
+
+This is a first step towards full State Archival, which will lower the storage requirements of validators
+and decrease the growth of History Archives. Additionally, this opens the door for database optimizations of
+live Soroban state, increasing limits and throughput, as well as automatic entry restoration.
+
+### Goals Alignment
+
+This change is aligned with the goal of lowering the cost and increasing the scale of the network.
+
+## Abstract
+
+Currently both live and archived Soroban state are maintained in a single BucketList DB. This makes data prioritization difficult
+since all state exists in a singular data structure. This proposal separates archived state and live state into two separate DBs.
+While these DBs are still both persisted on disk by validators, optimizing live state access is significantly easier. In particular,
+live state can be entirely cached in memory, removing disk reads from the Soroban transaction execution path, greatly increasing
+read limits and throughput.
+
+Each validator will maintain two BucketLists, the Live BucketList and the Hot Archive BucketList. `PERSISTENT` Soroban
+data entries and code entries that have expired will be "evicted" from the Live BucketList and moved to the Hot Archive
+BucketList. Both the Live BucketList and Hot Archive BucketList are persisted on disk and written to History Archives.
+
+Note that this proposal is a subset of [CAP-0057](cap-0057.md). The Hot Archive here is the same as CAP-0057, and this CAP
+is a strict subset of CAP-0057. The purpose is not to replace CAP-0057, but to offer a smaller first step implementation wise.
+
+## Specification
+
+### XDR changes
+
+```
```

Comment:
This is usually rendered as a `diff` to make it easier to read :thinking:

Reply:
Yeah, I just have it in this format for now since I haven't actually made the XDR changes yet (since we're still finalizing them). Once we have consensus I'll update the actual XDR and provide a nice diff.